### DISCUSSION TITLE
Password Generator App

### DISCUSSION THREAD

**Discussion by jerryajeh** [TheAlgorithms/Python#11523] [General]
I am trying to build an APP that randomly generate password from only numbers, alphabet and characters using python framework. I am getting some errors

import random 

letters = ["A","B","C","D","E","F","G","H","I","J","K","L","M","N","O","P","Q","R","S","T","U","V","W","X","Y","Z",
           "a","b","c","d","e","f","g","h","i","j","k","l","m","n","o","p","q","r","s","t","u","v","w","x","y","z"]
numbers = ["0","1","2","3","4","5","6","7","8","9"]
symbols = ["!","@","+","$","#","&","*","-"]

print("Welcome to my Password Generator App")
print()
n_letters = int(input("How many letters do you like in your password:\n"))
n_numbers = int(input("How many numbers do you like in your password:\n"))
n_symbols = int(input("How many symbols do you like in your password:\n"))

password = ""
password_list = []

for i in range(1,n_letters + 1):
    password_list.append(random.choice(letters))

for i in range(1,n_numbers + 1):
    password_list.append(random.choice(numbers))

for i in range(1,n_symbols + 1):
    password_list.append(random.choice(symbols))

for i in password_list:
    password = password + i

print(password_list)

**Comment by par-swa** [comment]
Hello, jerry, you are missing indentation after for loop, and instead of print(password_list) give print(''.join(password_list)) 
so that it will give you as a word

**Reply by Latheeshroy06** [reply]
yes u are right we cant edit in that code right after submission know

**Reply by SatoshiMamoto** [reply]
шкибиди тойлет доб доб доб ты за камеро мено567 туалетов? print = gaga

**Comment by imSanko** [comment]
```
import random
import string

def password_generator():
    print("Welcome to my Password Generator App\n")

    letters = string.ascii_letters
    numbers = string.digits
    symbols = string.punctuation

    n_letters = int(input("How many letters do you like in your password:\n"))
    n_numbers = int(input("How many numbers do you like in your password:\n"))
    n_symbols = int(input("How many symbols do you like in your password:\n"))

    password_list = [
        *random.choices(letters, k=n_letters),
        *random.choices(numbers, k=n_numbers),
        *random.choices(symbols, k=n_symbols)
    ]

    random.shuffle(password_list)

    password = "".join(password_list)

    print("Generated Password: ", password)
    print("Password List: ", password_list)

password_generator()
```

@jerryajeh 

-----------**MARK IT AS ANSWERD IF IT HELPS**-----------

**Reply by CodingwithFawad** [reply]
❤️

**Comment by McodexD** [comment]
❤️

**Reply by CodingwithFawad** [reply]
❤️
